cmd/sonobuoy/app/gen.go

```diff
@@ -65,14 +69,46 @@ func (g *genFlags) Config() (*client.GenConfig, error) {
 		return nil, errors.Wrap(err, "could not retrieve E2E config")
 	}
 
+	kubeclient, kubeError := maybeGetClient(&g.kubecfg)
+
+	rbacEnabled, err := genflags.rbacMode.Enabled(kubeclient)
+	if err != nil {
+		if errors.Cause(err) == ErrRBACNoClient {
+			return nil, errors.Wrap(err, kubeError.Error())
+		}
+		return nil, err
+	}
+
+	var discoveryClient discovery.ServerVersionInterface
+	var image string
+
+	if g.kubeConformanceImage != "" {
```

Comment:
This part could be at the very top, if kubeConformanceImage is defined, return the config. Otherwise, we can do the client song and dance and then return the config. This would drop the indentation level, be easier to read, and remove some of the error checking here I think.

Reply:
We also need the client song and dance for RBAC detection.

Reply:
ughhhhh yeah true